File: services/sphana-data-store-service/src/python/sphana_data_store/controllers/indices/v1/schemas/create_index_request.py
from typing import Optional
from pydantic import BaseModel

class CreateIndexRequest(BaseModel):
    index_name: Optional[str] = None
    description: Optional[str] = None
    media_type: Optional[str] = None
    dimension: Optional[int] = None
    number_of_shards: Optional[int] = None

File: services/Sphana/Services/Sphana.DataStore/Presentation/Schemas/Indices/CreateIndexRequest.cs
using System.Text.Json.Serialization;

namespace Sphana.DataStore.Presentation.Schemas.Indices;

public sealed class CreateIndexRequest
{
    [JsonPropertyName("index_name")]
    public required string IndexName { get; init; }

    [JsonPropertyName("description")]
    public required string Description { get; init; }

    [JsonPropertyName("media_type")]
    public required string MediaType { get; init; }

    [JsonPropertyName("dimension")]
    public required int Dimension { get; init; }

    [JsonPropertyName("number_of_shards")]
    public required int NumberOfShards { get; init; }
}